framework/include/meshgenerators/ParsedElementDeletionGenerator.h

```diff
@@ -0,0 +1,38 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementDeletionGeneratorBase.h"
+#include "FunctionParserUtils.h"
+
+namespace libmesh
+{
+class Point;
+}
+
+/**
+ * Deletes elements based on the evaluation of a parsed expression, involving the coordinates of
+ * their vertex average and their volume
+ */
+class ParsedElementDeletionGenerator : public ElementDeletionGeneratorBase,
```

Comment:
It's an existing `ElementDeletionGeneratorBase` issue, nothing that needs to be fixed in this PR, but:

If we delete a child element of a libMesh mesh without also deleting its parent, or vice-versa, the resulting mesh is invalid, almost certain to fail assertions in `devel` or segfault in `opt`.  And choosing based on element volume, where parents will exceed their children by `2^d`, seems especially likely to trigger that.  This scares me.

I'm having trouble picturing the use case for this too.  Do we want to remove part of our domain?  Generally an efficient FEM analysis should use smaller elements in the most important parts of the domain; why are smaller elements in particular the parts we want to instead remove entirely?

Reply:
I ll take a look separately at this child/parent deletion.

I have a bad mesh with tiny and bad elements on the boundaries. Deleting them lets the simulation go forward but does not inflence results afaicansee

At the end of the day the mesh should be redone, it s not my decision to make

Reply:
Would it make sense to select for deletion based on something other than `Elem::volume()`?  `Elem::quality(ASPECT_RATIO)` or another quality measure?  There's no real a-priori way to define "bad elements" (e.g. flow simulations often grade sharply into boundaries, so the final layer of elements is tiny *and* high aspect-ratio) but in general bad shape tends to screw up conditioning in a less preconditioner-recoverable way than bad size does.

Would it make sense to filter ... somehow ... for elements that are on the boundary?  It sounds like the answer to "Do we want to remove part of our domain?" is "Yes!", but it's a case where we can get away with removing boundary elements specifically because that just shrinks the domain a tiny bit; accidentally adding gaps mid-domain might be more serious.  The trouble is, I can't figure out how to do this properly; the obvious algorithm would require reimplementing half of the base class.

I guess this works as-is, as a "the mesh should be redone but it won't be" hack.

Reply:
but actually following that same logic of "this is my need right now" I added the element id. With this, I should be able to finish torturing Kyle's mesh and say "if you do not redo your mesh or pay for fancier libmesh smoothing, this is as good as I can get that mesh to be"